perun-registrar-lib/src/main/java/cz/metacentrum/perun/registrar/impl/ExpirationNotifScheduler.java

```diff
@@ -357,6 +378,82 @@ private void checkVoMembersState(List<Vo> vos) throws WrongAttributeAssignmentEx
 		expireSponsorships();
 	}
 
+	/**
+	 * Checks all applications for given vos and if some application is expired (according to expiration rules set by
+	 * VO Manager), then rejects this application.
+	 *
+	 * @param vos virtual organizations
+	 * @throws PerunException
+	 */
+	private void voApplicationsAutoRejection(List<Vo> vos) throws PerunException {
+		List<String> states = new ArrayList<>();
+		states.add("NEW");
+		states.add("VERIFIED");
+
+		for (Vo vo : vos) {
+			Attribute expiration = perun.getAttributesManagerBl().getAttribute(sess, vo, A_VO_APP_EXP_RULES);
+			if (expiration != null) {
+				List<Application> applications = registrarManager.getApplicationsForVo(sess, vo, states);
+				rejectExpiredApplications(applications, expiration);
+			}
+		}
+	}
+
+	/**
+	 * Gets all existing groups and then checks all applications for this groups and if some application is expired
+	 * (according to expiration rules set by VO Manager), then rejects this application.
+	 *
+	 * @param vos virtual organizations
+	 * @throws PerunException
+	 */
+	private void groupApplicationsAutoRejection(List<Vo> vos) throws PerunException {
+		List<String> states = new ArrayList<>();
+		states.add("NEW");
+		states.add("VERIFIED");
+
+		List<Group> groups = new ArrayList<>();
+		for (Vo vo : vos) {
+			groups.addAll(perun.getGroupsManagerBl().getGroups(sess, vo));
+		}
+
+		for (Group group : groups) {
+			Attribute expiration = perun.getAttributesManagerBl().getAttribute(sess, group, A_GROUP_APP_EXP_RULES);
+			if (expiration != null) {
+				List<Application> applications = registrarManager.getApplicationsForGroup(sess, group, states);
+				rejectExpiredApplications(applications, expiration);
+			}
+		}
+	}
+
+	/**
+	 * Compares creation day of application to values in expiration attribute and if find expired application, then
+	 * reject it.
+	 *
+	 * @param applications applications
+	 * @param expiration attribute with number of days to application expiration
+	 * @throws PerunException
+	 */
+	private void rejectExpiredApplications (List<Application> applications, Attribute expiration) throws PerunException {
+		Map<String, String> attrValue = expiration.valueAsMap();
+		for(Application application : applications) {
+			String date = application.getCreatedAt();
+			LocalDate createdAt = LocalDate.parse(date.substring(0, 10));
+			LocalDate now = getCurrentLocalDate();
+			if (application.getState().toString().equals("NEW")) {
```

Comment:
For enums, you can compare values directly:
```suggestion
			if (application.getState() == AppState.NEW)) {
```

Reply:
Changed